### PR TITLE
doc: renew tutorial, suggest passive client, restructure heartbeat interval

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
This PR aims add several references to point people to passive mode for client

The passive mode has many advantages when backing up behind NAT Device or Firewalls, also in other subnets:

- The client does not need to be able to resolve the storage name
- The client does not need to connect to the storage
- The client only recieves incoming connections

Thus, we should emphasize this both in the "add new client" chapter and in the troubleshooting chapter to try passive mode in case the backups do not work. There's already an existing chapter describing TasksAndConcepts/NetworkSetup.html#passive-clients, so we add links to it from several other location.
OP#5505

The second part of the PR is renewing the `HeartbeatInterval` parameter and the associated description.
A common section is added to the Network Chapter with indexes and keyword for a better search, (including when searching firewall ). All manually_added_config_directive_descriptions/*HeartbeatInterval.rst.inc have been cleaned up and reference the common section.
OP#5384

The third part is fixing the config-Dir_Director_MaximumConcurrentJobs informations 
OP#5547

#### Please check

- [X] Short description and the purpose of this PR is present _above this paragraph_
- [X] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

**Review by pstorz — CHANGES_REQUESTED** [review]
Good work! Please see comments.

File: docs/manuals/source/Appendix/Troubleshooting.rst [hunk 1]
```diff
@@ -26,7 +26,7 @@ There are several reasons why a |dir| could not contact a client on a different
 
 -  Your password or names are not correct in both the Director and the Client machine. Try configuring everything identical to how you run the client on the same machine as the Director, but just change the address. If that works, make the other changes one step at a time until it works.
 
-Some of the DNS and Firewall problems can be circumvented by configuring clients using :ref:`section-ClientInitiatedConnection` or as :ref:`PassiveClient`.
+Some of the DNS and Firewall problems can be circumvented by configuring clients using :ref:`section-ClientInitiatedConnection` or as :ref:`section-PassiveClient`.
```
[the remark this answers is not in the document]
Comment (bruno-at-bareos) [reply docs/manuals/source/Appendix/Troubleshooting.rst:29]: right.

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst [hunk 2]
```diff
@@ -310,22 +323,21 @@ Here we have the choice to run (yes), to modify one or more of the above paramet
...
    :caption: run
 
    *<input>messages</input>
-   28-Apr-2003 14:30 bareos-sd: Wrote label to prelabeled Volume
+   28-Apr-2019 14:30 bareos-sd: Wrote label to prelabeled Volume
       "TestVolume001" on device /var/lib/bareos/storage
-   28-Apr-2003 14:30 rufus-dir: Bareos 1.30 (28Apr03): 28-Apr-2003 14:30
+   28-Apr-2019 14:30 bareos-dir: Bareos 18.2 (28Apr19): 28-Apr-2019 14:30
    JobId:                  1
    Job:                    BackupClient1.2003-04-28_14.22.33
```
Comment (bruno-at-bareos) [inline docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:335]:
```suggestion
   Job:                    BackupClient1.2019-04-28_14.22.33
```

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst [hunk 3]
```diff
@@ -310,22 +323,21 @@ Here we have the choice to run (yes), to modify one or more of the above paramet
...
 If you wait a few seconds, then enter the command :bcommand:`messages` you will get back something like:
 
-.. TODO: Replace bconsole output by current version of Bareos.
 
 .. code-block:: bconsole
    :caption: run
 
    *<input>messages</input>
-   28-Apr-2003 14:30 bareos-sd: Wrote label to prelabeled Volume
+   28-Apr-2019 14:30 bareos-sd: Wrote label to prelabeled Volume
```
[the remark this answers is not in the document]
Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:331]: Ok I see them now :scream:  .. I will replace the whole by 22.1.0 version.

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst [hunk 4]
```diff
@@ -738,6 +761,12 @@ After modifying the file, you have to restart the |fd|:
 
    service bareos-fd restart
 
+.. note::
+
+   You should also consider using Passive mode for the the client,
+   see :ref:`section-PassiveClient`.
```
Comment (bruno-at-bareos) [inline docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:767]:
```suggestion
   see :ref:`section-PassiveClient`, especially if you are backing up clients that are not part of your local network.
```

[the remark this answers is not in the document]
Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/Tutorial.rst]: ok changed too

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst [hunk 5]
```diff
@@ -48,6 +51,14 @@ Assuming you have installed the packages, to start the three daemons, from your
    service bareos-sd start
    service bareos-fd start
 
+With systemd enable system it is even easier:
```
Comment (pstorz) [inline docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:54]:
```suggestion
With systemd enabled system it is even easier:
```

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst [hunk 6]
```diff
@@ -260,7 +335,7 @@ Now, let’s actually run a job with:
 .. code-block:: bconsole
    :caption: run
 
-   run
+   <imput>run</input>
```
Comment (pstorz) [inline docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:338]:
```suggestion
   <input>run</input>
```

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst [hunk 7]
```diff
@@ -275,77 +350,125 @@ you should get the following output:
...
 
 .. code-block:: bconsole
-   :caption: run
+   :caption: joblog output
 
    *<input>messages</input>
-   28-Apr-2003 14:30 bareos-sd: Wrote label to prelabeled Volume
-      "TestVolume001" on device /var/lib/bareos/storage
-   28-Apr-2003 14:30 rufus-dir: Bareos 1.30 (28Apr03): 28-Apr-2003 14:30
+   21-Sep 12:58 bareos-dir JobId 1: No prior Full backup Job record found.
```
Comment (pstorz) [inline docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:408]: We have 21-Sep everywhere here, but the End time is 23-Sep.

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst [hunk 8]
```diff
@@ -472,87 +630,92 @@ and you will get the following output:
...
-     FD termination status:  OK
-     SD termination status:  OK
-     Termination:            Restore OK
-
-
-
-After exiting the Console program, you can examine the files in :file:`/tmp/bareos-restores`, which will contain a small directory tree with all the files. Be sure to clean up at the end with:
+   :caption: restore job report
+
+   21-Sep 13:20 bareos-dir JobId 3: Start Restore Job RestoreFiles.2023-09-23_13.20.42_09
```
Comment (pstorz) [inline docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:680]: Here again we have 21-Sep but End Time of 23-Sep

File: docs/manuals/source/TasksAndConcepts/NetworkSetup.rst [hunk 9]
```diff
@@ -1,11 +1,53 @@
...
+   Consult `TLDP howto TCP-Keepalive <https://tldp.org/HOWTO/TCP-Keepalive-HOWTO/overview.html>`_
+   to learn how to setup tcp keepalive in you Linux.
+
+
 .. _section-ClientInitiatedConnection:
 
 Client Initiated Connection
 ---------------------------
 
+.. index::
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/NetworkSetup.rst:47]: Do we maybe want to put the Passive Client here, and the Client Initated connection after that? I think that the passive client is probably much more used than the client initiated connection.

**Review by pstorz — CHANGES_REQUESTED** [review]
Good work! Just see my small comments please!

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst (hunk 5 above)
Comment (bruno-at-bareos) [inline docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:54]:
```suggestion
With systemd enabled  it is even easier:
```

Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:54]: I will even change that a bit more.

File: docs/manuals/source/IntroductionAndTutorial/Tutorial.rst (hunk 7 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/Tutorial.rst:408]: ok that's a mistake should be 23-Sep my regexp missed that one format. Will be so easier with unified timestamps ;-)

File: docs/manuals/source/TasksAndConcepts/NetworkSetup.rst (hunk 9 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/TasksAndConcepts/NetworkSetup.rst:47]: Make sense, will add that change

**Review by pstorz — APPROVED** [review]
Good work!

**Pull request merged**